moveit_ros/planning/planning_request_adapter_plugins/src/chomp_optimizer_adapter.cpp

```diff
@@ -0,0 +1,246 @@
+                      << params_.trajectory_initialization_method_);
+    }
+  }
+
+  virtual std::string getDescription() const
+  {
+    return "CHOMP Optimizer !!!!";
+  }
+
+  virtual bool adaptAndPlan(const PlannerFn& planner, const planning_scene::PlanningSceneConstPtr& planning_scene,
+                            const planning_interface::MotionPlanRequest& req,
+                            planning_interface::MotionPlanResponse& res,
+                            std::vector<std::size_t>& added_path_index) const
+  {
+    // following call to planner() calls the OMPL planner and stores the trajectory inside the MotionPlanResponse res
+    // variable which is then used by CHOMP for optimization of the computed trajectory
+    bool solved = planner(planning_scene, req, res);
+
+    // create a hybrid collision detector to set the collision checker as hybrid
+    collision_detection::CollisionDetectorAllocatorPtr hybrid_cd(
+        collision_detection::CollisionDetectorAllocatorHybrid::create());
+
+    ROS_INFO_STREAM("Configuring Planning Scene for CHOMP ....");
+    planning_scene->setActiveCollisionDetector_ConstVersion(hybrid_cd, true);
+
+    chomp::ChompPlanner chompPlanner;
+    planning_interface::MotionPlanDetailedResponse res_detailed;
+    moveit_msgs::MotionPlanDetailedResponse res2;
+    moveit_msgs::MotionPlanRequest req_moveit_msgs;
+
+    // populate the trajectory to pass to CHOMPPlanner::solve() method. Obtain trajectory from OMPL's
+    // planning_interface::MotionPlanResponse object and put / populate it in the
+    // moveit_msgs::MotionPlanDetailedResponse object
+    moveit_msgs::RobotTrajectory trajectory_msgs2;
+    res.trajectory_->getRobotTrajectoryMsg(trajectory_msgs2);
+    res2.trajectory.resize(1);
+    res2.trajectory[0] = trajectory_msgs2;
+
+    bool planning_success;
+
+    bool temp = chompPlanner.solve(planning_scene, req, params_, res2);
+
+    if (temp)
+    {
+      res_detailed.trajectory_.resize(1);
+      res_detailed.trajectory_[0] = robot_trajectory::RobotTrajectoryPtr(
+          new robot_trajectory::RobotTrajectory(planning_scene->getRobotModel(), "panda_arm"));
+
+      moveit::core::RobotState start_state(planning_scene->getRobotModel());
+      robot_state::robotStateMsgToRobotState(res2.trajectory_start, start_state);
+      res_detailed.trajectory_[0]->setRobotTrajectoryMsg(start_state, res2.trajectory[0]);
+      res_detailed.description_.push_back("plan");
+      res_detailed.processing_time_ = res2.processing_time;
+      res_detailed.error_code_ = res2.error_code;
+      planning_success = true;
+    }
+    else
+    {
+      res_detailed.error_code_ = res2.error_code;
+      planning_success = false;
+    }
+
+    res.error_code_ = res_detailed.error_code_;
+
+    if (planning_success)
+    {
+      res.trajectory_ = res_detailed.trajectory_[0];
+      res.planning_time_ = res_detailed.processing_time_[0];
+    }
+
```

Comment:
If the initial trajectory is valid, but CHOMP makes it invalid, does the planning adaptor return the initial trajectory?

Reply:
Yes, it should as per [this block](https://github.com/raghavendersahdev/moveit/blob/CHOMP_planning_request_adapter/moveit_ros/planning/planning_request_adapter_plugins/src/chomp_optimizer_adapter.cpp#L231), if `planning_success` from CHOMP is `false`, then the `res MotionPlanDetailedResponse` object is never changed from CHOMP and should stay the same as obtained from OMPL as [this](https://github.com/raghavendersahdev/moveit/blob/CHOMP_planning_request_adapter/moveit_ros/planning/planning_request_adapter_plugins/src/chomp_optimizer_adapter.cpp#L233) is the only place where the response trajectory is changed in the chomp_optimizer_adapter.cpp apart from being set initially [here](https://github.com/raghavendersahdev/moveit/blob/CHOMP_planning_request_adapter/moveit_ros/planning/planning_request_adapter_plugins/src/chomp_optimizer_adapter.cpp#L183) in the OMPL planner.